ledger-core/virtual/integration/deduplication_test.go

```diff
@@ -99,9 +103,227 @@ func TestDeduplication_Constructor_DuringExecution(t *testing.T) {
+	var (
+		isolation = contract.ConstructorIsolation()
+		outgoing  = server.RandomLocalWithPulse()
+		class     = gen.UniqueGlobalRef()
+	)
+
+	pl := payload.VCallRequest{
+		CallType:       payload.CTConstructor,
+		CallFlags:      payload.BuildCallFlags(isolation.Interference, isolation.State),
+		Callee:         class,
+		CallSiteMethod: "New",
+		CallOutgoing:   outgoing,
+	}
+
+	synchronizeExecution := synchronization.NewPoint(1)
+	defer synchronizeExecution.Done()
+
+	{
+		requestResult := requestresult.New([]byte("123"), gen.UniqueGlobalRef())
+		requestResult.SetActivate(gen.UniqueGlobalRef(), class, []byte("234"))
+
+		executionMock := runnerMock.AddExecutionMock(calculateOutgoing(pl).String())
+		executionMock.AddStart(nil, &execution.Update{
+			Type:   execution.Done,
+			Result: requestResult,
+		})
+	}
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+		synchronizeExecution.Synchronize()
+		return false
+	})
+	typedChecker.VStateReport.SetResend(true)
+
+	{
+		server.SendPayload(ctx, &pl)
+	}
+
+	testutils.WaitSignalsTimed(t, 10*time.Second, synchronizeExecution.Wait())
+	server.IncrementPulse(ctx)
+	synchronizeExecution.WakeUp()
+
+	{
+		server.SendPayload(ctx, &pl)
+	}
+
+	testutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
+	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+
+	{
+		assert.Equal(t, 1, typedChecker.VCallResult.Count())
+		assert.Equal(t, 1, typedChecker.VStateReport.Count())
+	}
+
+	mc.Finish()
+}
+
+func TestDeduplication_Constructor_DuringExecution_SamePulse(t *testing.T) {
+	t.Log("C4998")
+
+	var (
+		mc = minimock.NewController(t)
+	)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 2)
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+
+	var (
+		isolation = contract.ConstructorIsolation()
+		outgoing  = server.RandomLocalWithPulse()
+		class     = gen.UniqueGlobalRef()
+	)
+
+	pl := payload.VCallRequest{
+		CallType:       payload.CTConstructor,
+		CallFlags:      payload.BuildCallFlags(isolation.Interference, isolation.State),
+		Callee:         class,
+		CallSiteMethod: "New",
+		CallOutgoing:   outgoing,
+	}
+
+	synchronizeExecution := synchronization.NewPoint(1)
+
+	{
+		requestResult := requestresult.New([]byte("123"), gen.UniqueGlobalRef())
+		requestResult.SetActivate(gen.UniqueGlobalRef(), class, []byte("234"))
+
+		executionMock := runnerMock.AddExecutionMock(calculateOutgoing(pl).String())
+		executionMock.AddStart(func(ctx execution.Context) {
+			synchronizeExecution.Synchronize()
+		}, &execution.Update{
+			Type:   execution.Done,
+			Result: requestResult,
+		})
+	}
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VCallResult.SetResend(false).ExpectedCount(100)
```

Comment:
expected 100, why?

Reply:
sorry, typo; was checking that feature with "ExpectedCount" works